src/Filter/FilterProcessor.php

```diff
@@ -0,0 +1,19 @@
+<?php
+/**
+ * This file is part of the event-engine/php-postgres-document-store.
+ * (c) 2019-2021 prooph software GmbH <contact@prooph.de>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace EventEngine\DocumentStore\Postgres\Filter;
+
+use EventEngine\DocumentStore\Filter\Filter;
+
+interface FilterProcessor
+{
+    public function process(Filter $filter): array;
```

Comment:
we should simplify the interface. `filterToWhereClause` was a private method before and returned a tuple with optional return values, because it was called recursively to return sub parts of a where clause.

The process function here should only return a `string` which is the compiled where clause. DefaultFilterProcess should continue to use an internal `filterToWhereClause` method that returns a tuple with optional return values.

Reply:
You're right. Returning an array is just plain terrible 😄 But not just a string must be returned, also arguments used by the query statement...

Reply:
:thinking: what about a value object containing query as a string + arguments array? At least this would be more clear than a top level array.